Use braces when expanding arrays

joehorsnell opened the pull request:
Fix the generation of zsh completion scripts to prevent [ShellCheck][1] warnings
around [using braces when expanding arrays][2].

Edit: Just for future reference, it's worth noting that ShellCheck [doesn't officially support zsh](https://github.com/koalaman/shellcheck/issues/809). I am very much _not_ a shell expert, so don't claim to know the differences between bash and zsh when it comes to the checks that ShellCheck performs, but I _think_ this change is still valid for zsh.

It might be worth instead adding the pragma to the generated file to tell ShellCheck not to run?

[1]: https://www.shellcheck.net/
[2]: https://github.com/koalaman/shellcheck/wiki/SC1087

joehorsnell committed:
Use braces when expanding arrays

Fix the generation of zsh completion scripts to prevent [ShellCheck][1] warnings
around [using braces when expanding arrays][2].

[1]: https://www.shellcheck.net/
[2]: https://github.com/koalaman/shellcheck/wiki/SC1087

labocho commented:
I didn't know ShellCheck. It's cool.
Thank you for your request!

joehorsnell commented:
> I didn't know ShellCheck. It's cool.
> Thank you for your request!

Thanks for the quick response @labocho. I've just realised, there are actually a couple of other ShellCheck warnings in the generated scripts, so I'll do another PR to fix them.

Would you be able to cut a new version to RubyGems once those fixes are in?

Edit: I can see you've already pushed [0.1.6](https://rubygems.org/gems/thor-zsh_completion/versions/0.1.6), thanks! I'll get the other fix PR open, then hopefully you can push another version? Thanks.

labocho commented:
Oh, I should have checked ShellCheck warnings myself.

Yes. I am waiting for another PR, and then release a new version of RubyGems.

joehorsnell commented:
> Oh, I should have checked ShellCheck warnings myself.
> 
> Yes. I am waiting for another PR, and then release a new version of RubyGems.

Ok cool. See also my edited description above re ShellCheck not officially supporting zsh. I am fairly sure the change in this PR is still valid, but it might be worth instead adding the pragma to the generated zsh script to tell ShellCheck to ignore the file completely?

labocho commented:
Thank you for the information. Unsupporting zsh is unfortunately, but it is still useful!

I fixed script, and added pragma for unavoidable point (50147bb03019cdef6ff49ac154e1e5a3a2a1df9f). Generated script is pass ShellCheck (with `--shell bash` option) and works.

joehorsnell commented:
> Thank you for the information. Unsupporting zsh is unfortunately, but it is still useful!
> 
> I fixed script, and added pragma for unavoidable point ([50147bb](https://github.com/labocho/thor-zsh_completion/commit/50147bb03019cdef6ff49ac154e1e5a3a2a1df9f)). Generated script is pass ShellCheck (with `--shell bash` option) and works.

That's great, thanks @labocho.

Are you able to do a new version 0.1.7 with you fix 50147bb03019cdef6ff49ac154e1e5a3a2a1df9f in it?

Cheers.

labocho commented:
> Are you able to do a new version 0.1.7 with you fix 50147bb in it?

Yes. Released 0.1.7!
https://rubygems.org/gems/thor-zsh_completion/versions/0.1.7

joehorsnell commented:
Thanks @labocho!
